build/rpmfc.c

```diff
@@ -1184,13 +1184,15 @@ static int initAttrs(rpmfc fc)
 
     /* Discover known attributes from pathnames + initialize them */
     if (rpmGlob(attrPath, NULL, &files) == 0) {
-	nattrs = argvCount(files);
+	int nfiles = argvCount(files);
+	nattrs = nfiles + 1;
 	fc->atypes = xcalloc(nattrs + 1, sizeof(*fc->atypes));
-	for (int i = 0; i < nattrs; i++) {
+	for (int i = 0; i < nfiles; i++) {
 	    char *bn = basename(files[i]);
 	    bn[strlen(bn)-strlen(".attr")] = '\0';
 	    fc->atypes[i] = rpmfcAttrNew(bn);
 	}
+	fc->atypes[nattrs - 1] = rpmfcAttrNew("local_generator");
```

Comment:
The `nfiles` could be used here instead, but I am not sure it would make the code more readable.